### DISCUSSION TITLE
Typescript refactoring across files doesn't work

### DISCUSSION THREAD

**Discussion by baronyoung** [LazyVim/LazyVim#4204] [Q&A] [answered]
Using the standard typescript lsp enabled via mason and Extras, if I rename a typescript type that exists in one file, it doesn't properly refactor in other files.  For example, if I have a "Person" type with 'name' and 'age' as properties, a variable in a separate file which uses this type will not change it's property names if I change it in the type definition.

Am I missing a plugin or config setting somehwere?  ChatGPT says use 'coc-tsserver' but that doens't sound right.

**Comment by rogeriofrsouza** [comment]
Are you using `<leader>cr` to rename the field/property?
Just tried here, seems like it's working fine.


https://github.com/user-attachments/assets/25b5ba33-4b45-4b29-b731-dd9699a1cfe6

**Reply by baronyoung** [reply]
I was expecting it to work without the other file having to be open.  For large projects I don't think this is practical unless I'm missing something.
And to answer your question, yes, I'm using leader-cr

**Reply by rogeriofrsouza** [reply]
Oh, the other files buffers were open just for convenience. It's not required

**Comment by baronyoung** [comment]
You're correct.  I just tried it in an actual project directory instead of a temp directory and it did work correctly.  I'm not sure why it didn't work before but thanks for your help!  I'll close this.

**Reply by rogeriofrsouza** [reply, marked as answer]
Ok, I was able to reproduce it in a directory without project config files in the root.

[nvim-lspconfig default configuration](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#vtsls) for `vtsls` states that:

> To configure a TypeScript project, add a [tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) or [jsconfig.json](https://code.visualstudio.com/docs/languages/jsconfig) to the root of your project.

And the LSP will recognize the root of your TypeScript project if it has some specific files:

> root_pattern("tsconfig.json", "package.json", "jsconfig.json", ".git")

**Reply by baronyoung** [reply]
I suspected it was probably something like that.  It's good to know the specific reason.  Thanks for taking the time to share this!
